examples/gno.land/p/demo/grc/grc20/grc20.gno

```diff
@@ -0,0 +1,234 @@
+
+	return true
+}
+
+func (t *Token) Allowance(owner, spender std.Address) (remaining uint64) {
+	rawAllowance, found := t.allowances.Get(spenderKey(owner, spender))
+	if !found {
+		return 0
+	}
+
+	return rawAllowance.(uint64)
+}
+
+func (t *Token) Mint(account std.Address, amount uint64) {
+	mustBeValid(account)
+	t.update(emptyAddress, account, amount)
+}
+
+func (t *Token) Burn(account std.Address, amount uint64) {
+	mustBeValid(account)
+	t.update(account, emptyAddress, amount)
+}
+
+// Helpers
+func (t *Token) update(from, to std.Address, value uint64) {
+	// If new tokens are minted, check for overflow
+	if from == emptyAddress {
+		sum, carry := bits.Add64(t.totalSupply, value, 0)
+		if carry != 0 {
+			err := ufmt.Sprintf("GRC20($%s): uint64 overflow: %d + %d", t.symbol, t.totalSupply, value)
+			panic(err)
+		}
+
+		t.totalSupply = sum
+	} else {
+		// Deduct `value` from `from`
+		rawFromBalance, found := t.balances.Get(from.String())
+		if !found {
+			err := ufmt.Sprintf("GRC20($%s): Address %s not found", t.symbol, from)
+			panic(err)
+		}
+
+		fromBalance := rawFromBalance.(uint64)
+
+		if fromBalance < value {
+			err := ufmt.Sprintf("GRC20($%s): Insufficient balance %s, %s, %s", t.symbol, fromBalance, from, value)
+			panic(err)
+		}
+
+		// Overflow not possible: value <= fromBalance <= totalSupply
+		t.balances.Set(from.String(), fromBalance-value)
+	}
+
+	// Check if coins are burned
+	if to == emptyAddress {
+		// Overflow not possible: value <= totalSupply or value <= fromBalance <= totalSupply
+		t.totalSupply -= value
+	} else {
+		// Valid transfer
+		rawToBalance, found := t.balances.Get(to.String())
+		if !found {
+			// Initialize the balance the first time the address receives tokens
+			t.balances.Set(to.String(), value)
+		} else {
+			toBalance := rawToBalance.(uint64)
+			// Overflow not possible: balance + value is at most totalSupply, which we know fits into a uint64
+			t.balances.Set(to.String(), toBalance+value)
+		}
+	}
+
+	std.Emit(TransferEvent,
+		"from", from.String(),
+		"to", to.String(),
+		"value", strconv.Itoa(int(value)),
+	)
+}
+
+func (t *Token) transfer(from, to std.Address, value uint64) {
+	mustBeValid(from)
+	mustBeValid(to)
+	t.update(from, to, value)
+}
+
+func (t *Token) approve(owner, spender std.Address, value uint64, emit bool) {
+	mustBeValid(owner)
+	mustBeValid(spender)
+
+	t.allowances.Set(spenderKey(owner, spender), value)
+
+	if emit {
+		std.Emit(ApprovalEvent,
+			"owner", owner.String(),
+			"spender", spender.String(),
+			"value", strconv.Itoa(int(value)),
+		)
+	}
+}
+
+// spendAllowance updates `owner`s allowance for `spender` based on spent `value`
+// Does not update the allowance value in case of infinite allowance
+// Panics if not enough allowance is available
+// Does not emit an Approval event
+func (t *Token) spendAllowance(owner, spender std.Address, value uint64) {
+	currentAllowance := t.Allowance(owner, spender)
+
+	if currentAllowance != math.MaxUint64 {
```

Comment:
This exception should be documented. If I'm not mistaken, you consider a MaxUint64 allowance to be equivalent to an infinite allowance, correct?